extensions/gitpod-remote/src/portViewProvider.ts

```diff
@@ -0,0 +1,138 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import { GitpodExtensionContext, ExposedServedGitpodWorkspacePort, GitpodWorkspacePort, isExposedServedGitpodWorkspacePort, isExposedServedPort, PortInfo, TunnelDescriptionI } from 'gitpod-shared';
+import { PortsStatus } from '@gitpod/supervisor-api-grpc/lib/status_pb';
+
+const PortCommands = <const>['tunnelNetwork', 'tunnelHost', 'makePublic', 'makePrivate', 'preview', 'openBrowser', 'retryAutoExpose', 'urlCopy', 'queryPortData'];
+
+type PortCommand = typeof PortCommands[number];
+
+export class GitpodPortViewProvider implements vscode.WebviewViewProvider {
```

Comment:
Duplicate this for now as it's using configcat code on gitpod-web and migration is almost done, so not worth to move more code I think

Reply:
@mustard-mh Is it done can we just remove old code with this PR?